integration_tests/__tests__/test_environment.test.js

```diff
@@ -19,5 +22,12 @@ it('respects testEnvironment docblock', () => {
   const result = runJest.json('test-environment').json;
 
   expect(result.success).toBe(true);
-  expect(result.numTotalTests).toBe(1);
+  expect(result.numTotalTests).toBe(2);
+});
+
+it('triggers setup/teardown hooks', () => {
```

Comment:
Can you split this out to a new test file entirely, so that we don't have to skip on windows for the existent test in here?

Reply:
Just create a new `integration_tests/__tests__/async_test_environment.test.js`

Reply:
I would like to just skip the one that doesn't work in windows but not sure what's the best way of doing it. (The method skipOnWindows.test() isn't used anywhere in the project)
Maybe Jest can use an API for this?

Reply:
I would like to just skip the test but doesn't know the best way of doing it.
```
it('triggers setup/teardown hooks', () => {
  if (skipOnWindows.test()) {
    expect(true).toBe(true);
  } else {
    fs.writeFileSync('/tmp/teardown', '');
    runJest('test-environment');
    const teardown = fs.readFileSync('/tmp/teardown', 'utf8');
    expect(teardown).toBe('teardown');
  }
});
```
Maybe Jest can use an Api for this? (conditional skip on describe/test)

Reply:
just move it into a completely separate file, and you can continues doing `skipOnWindows.suite();`.

I want the file `integration_tests/__tests__/test_environment.test.js` to have no diff in this PR 🙂